app/Http/Controllers/Api/LostMaterialController.php

```diff
@@ -5,61 +5,47 @@
 use App\Http\Controllers\Controller;
 use Illuminate\Http\Request;
 use App\Material;
+use App\Http\Requests\StoreLostMaterial;
+use App\Http\Requests\UpdateLostMaterial;
 
 class LostMaterialController extends Controller
 {
+    private Material $material;
+
+    public function __construct(Material $material)
+    {
+        $this->material = $material;
+    }
+
     public function index(Request $request)
     {
-        $returned = $request->query('returned');
+        $filter = $request->query('returned');
         $materials = null;
 
-        if ((is_null($returned)) || ($returned !== "true" && $returned !== "false")) {
-            $materials = Material::all();
+        if ($this->isValidFilter($filter)) {
+            $materials = $this->material::all();
         } else {
-            $materials = Material::when($returned, function ($query, $returned) {
-                            return ($returned === "true")
-                                ? $query->whereNotNull('tooker_registration_mark')
-                                : $query->whereNotNull('returner_registration_mark')
-                                        ->whereNull('tooker_registration_mark');
-                        })
-                        ->get();
+            $materials = $this->material->filter($filter);
         }
 
         return response()->json($materials);
     }
 
-    public function store(Request $request)
+    private function isValidFilter(?string $filter)
     {
-        $validatedData = $request->validate([
-            'name' => 'string|required',
-            'description' => 'string|required',
-            'returner_registration_mark' => 'string|required|max:60'
-        ]);
+        return (! $filter) || ($filter !== "true" && $filter !== "false");
```

Comment:
It means that ```filter = false``` pass? Or any other value that is equals to false?

Reply:
It means that null or any value different than true or false will return all the materials. If true it will return only the lost materials returned to the owner, and if false the other ones. Now I see that I forget to return all lost materials. I will fix it.

Reply:
Ok, my suggest is to rename the method isValidFilter to isNotValidFilter. :smile:

Reply:
Ok